frontend/src/Components/LibraryCard.tsx

```diff
@@ -57,6 +88,16 @@ export default function LibraryCard({
                 </figure>
                 <h3 className="w-3/4 body my-auto">{library.name}</h3>
             </div>
+
+            {role === UserRole.Student && (
+                <div
+                    className="tooltip tooltip-top absolute right-1 top-2 w-6 h-6 z-100"
+                    data-tip="Favorite video"
+                    onClick={(e) => e.stopPropagation()}
+                >
+                    {favoriteIcon}
+                </div>
```

Comment:
Does using <ULIComponent> interfere with this? If not, for consistency could you implement this using that component? It just wraps the icon with the tooltip and you'd have to pass in the extra classes in there. I think it should probably be updated anywhere where we use the favorite icon, so make sure to update the other pages too.

Reply:
Everything else I fixed, but this is confusing me?

Reply:
like it could maybe look something like this?
```
                <div onClick={(e: MouseEvent) => toggleFavorite(e)}>
                    <ULIComponent
                        tooltipClassName={'absolute right-2 top-2 z-100'}
                        iconClassName={`w-6 h-6 ${library.is_favorited ? 'text-primary-yellow' : ''}`}
                        icon={library.is_favorited ? StarIcon : StarIconOutline}
                        dataTip="Favorite Library"
                    />
                </div>
```
and inside of togglefavorite you'd just have to call `e.stopPropogation( )` - I don't think you need `e.preventDefault()` since I don't think this element has any default behavior?

Also- this tooltip should be favorite library rather than favorite video :)

Reply:
I have the `onClick` on a div outside of it since the `ULIComponent`'s `onClick` doesn't accept a param, although maybe we could update it to optionally accept a mouse event to be able to pass in that function into the `ULIComponent` so we dont have to have a wrapper div?

Reply:
Cool, thanks a ton for the clarification, still trying to figure out all of this typescript and react stuff...